sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetIOSuite.scala

```diff
@@ -1585,6 +1585,17 @@ class ParquetIOSuite extends QueryTest with ParquetTest with SharedSparkSession
       }
     }
   }
+
+  test("SPARK-49991: Respect 'mapreduce.output.basename' to generate file names") {
+    withTempPath { dir =>
+      withSQLConf("mapreduce.output.basename" -> "apachespark") {
+        spark.range(1).coalesce(1).write.parquet(dir.getCanonicalPath)
+        val df = spark.read.parquet(dir.getCanonicalPath)
+        assert(df.inputFiles.head.contains("apachespark"))
```

Comment:
Can we double check if there aren't anything specific to `part-` files in our codebase?

e.g., 

```
core/src/main/scala/org/apache/spark/rdd/ReliableCheckpointRDD.scala:      .filter(_.getName.startsWith("part-"))
core/src/main/scala/org/apache/spark/rdd/ReliableCheckpointRDD.scala:      .sortBy(_.getName.stripPrefix("part-").toInt)
core/src/main/scala/org/apache/spark/rdd/ReliableCheckpointRDD.scala:    "part-%05d".format(partitionIndex)
```

Reply:
For checkpoint* you have mentioned, it looks safe to me in a roundtrip.


In addition to those mentioned, there is another instance in SparkHadoopWriter where it is hardcoded. Since this affects the underlying RDD APIs, I plan to leave it unchanged in this PR.

Reply:
@mridulm @steveloughran wdyt?

Reply:
Very late LGTM :)